### DISCUSSION TITLE
rounded-full class doesn't completly rounds Image

### DISCUSSION THREAD

**Discussion by JyotiRSharma** [tailwindlabs/tailwindcss#12313] [Help] [answered]
Hiya! Next.JS is awesome!
However I came across a problem today while trying to round an image completely using the `rounded-full` class.

It just rounded the edges instead of making the entire image circular:
![image](https://github.com/tailwindlabs/tailwindcss/assets/28672278/71317121-ac33-4e11-b2e5-0e5f7c8526e4)

The code goes like this:
```JS
<Image
        src="https://hips.hearstapps.com/hmg-prod/images/wolf-dog-breeds-siberian-husky-1570411330.jpg"
        width={200}
        height={200}
        objectFit="cover"
        className='rounded-full'
        alt={"Image"}
      />
```

Isn't this code supposed to round the image completely like a circle?

But I did a workaround of using `style={{clipPath: 'circle()'}}` to circle the image:
```JS
<Image
        src="https://hips.hearstapps.com/hmg-prod/images/wolf-dog-breeds-siberian-husky-1570411330.jpg"
        width={200}
        height={200}
        objectFit="cover"
        style={{borderRadius: '9999px'}}
        alt={"Image"}
      />
```
![image](https://github.com/tailwindlabs/tailwindcss/assets/28672278/61102c3c-6b68-403c-8ada-227030ce50e6)

Am I doing something wrong? or Is `rounded-full` class working unexpectedly?

**Comment by wongjn** [comment, marked as answer]
`rounded-full` applies `border-radius: 9999px`. This is how `border-radius` applies with non-square elements – it will round as much as it can until the radius of the two corners meet up. Since the height is less than the width, the rounded corners meet up halfway vertically, which is less than the width.

To produce a circle with `rounded-full` (`border-radius: 9999px`), you'd need to ensure the image is square.

**Reply by JyotiRSharma** [reply]
Thank you very much!
